### PR TITLE
Feature/buildgcp

### PR DESCRIPTION
**Pull request by guiadco**
### Current behaviour

Actually all actiniumbox are upload manually, and the team couldn't keep it up-to-date

---
### Expected behaviour

As a user a want an actiniumbox up-to-date to test AllSpark installation.

---
### Modifications

> What are the changes involved to match with the expected behaviour ?

-  [x] Add vagrant local build template
-  [x] Add vagrant-cloud template
-  [x] Update CircleCI Config to validate the template 
-  [x] Update Circle Config to provision GCP and build the vagrant box under KVM Hypervisor and upload it in Vagrant Cloud
-  [x] Update CircleCI Config to Destroy GCP environnement
-  [x] Update CircleCI Config to shedule the workflow to build the vagrant box each month
-  [x] Update CircleCI Config to add manual update by updating the actiniumbox code
-  [x] Update Readme
-  [x] Add CONTRIBUTING
-  [x] First OS Build is CentOS 7.5.1804

---
### Status

- [x] Implementation
- [x] Test coverage
- [x] Documentation

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,17 +1,134 @@
...
   version: 2
-  validate:
+  validate-vagrantbox:
+    jobs:
+      - validate-vbox
+      - validate-kvm
+  schelude-build-kvm-vagrantbox:
+    triggers:
+      - schedule:
+          cron: "* 1 1 * *"
```
[the remark this answers is not in the document]
Comment (guiadco) [reply .circleci/config.yml]: No it's a mistake. the god cron is "0 1 1 * *"

**Comment by guiadco** [comment]
Ok to centralize builders.
Normally, i would do it in another PR but it could be done here.

**Comment by guiadco** [comment]
I will see this part in another PR

```
However, i don't know if there is a way to run post-processors conditionaly,
so we might need to keep the remote and local versions. I will take a look at it tomorrow.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - Make a Pull Request with :
+    - The new local and GCP Template for example: `ubuntu14.json & ubuntu14-local.json`
+    - Adapt the CircleCI Config to build the box automatically with the CI/CD
+      - Add command to add new GCP VM for example: `actinium-ubuntu14-box`
+      - Add Job to build the box
+      - Add to the Schedule Workflow and PR Workflow
+      - Add Validate Template to the job and the Workflow
+    - Add the build-<DistributionMajorversion>.sh script to build your OS for example `build-ubuntu14.sh`
+
+> The convention naimg is : '<DistributionMajorversion>' like centos7 or ubuntu14
```
Comment (Xide) [inline CONTRIBUTING.md]: s/naimg/name

File: centos7-local.json [hunk 3]
```diff
@@ -17,6 +14,50 @@
...
+      "format": "{{user `format`}}",
+      "accelerator": "kvm",
+      "net_device": "virtio-net",
+      "disk_interface": "virtio",
+      "http_directory": "http",
+      "boot_command": [
+        "<tab> text ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/{{user `ks_file`}}<enter><wait>"
+      ],
+      "shutdown_command": "echo 'vagrant'|sudo -S /sbin/halt -h -p",
+      "qemuargs": [[ "-m", "2048M" ],[ "-smp", "2" ]],
```
Comment (Xide) [inline centos7-local.json]: uniformize RAM between qemu / vbox

File: centos7.json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      "format": "{{user `format`}}",
+      "accelerator": "kvm",
+      "net_device": "virtio-net",
+      "disk_interface": "virtio",
+      "http_directory": "http",
+      "boot_command": [
+        "<tab> text ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/{{user `ks_file`}}<enter><wait>"
+      ],
+      "shutdown_command": "echo 'vagrant'|sudo -S /sbin/halt -h -p",
+      "qemuargs": [[ "-m", "2048M" ],[ "-smp", "2" ]],
```
Comment (Xide) [inline centos7.json]: uniformize qemu / vbox

File: CONTRIBUTING.md (hunk 2 above)
Comment (guiadco) [reply CONTRIBUTING.md]: Convention naming or Convention Name ?

**Comment by Xide** [comment]
lgtm

**Comment by Xide** [comment]
oops, forgot to rebase before the merge :/
Are you planning to create another branch today ? If not, we can always rewrite master history.

**Comment by guiadco** [comment]
CircleCI Fix done

**Pull request merged**